tests/phpunit/test-wc-stripe-sub-renewal.php

```diff
@@ -54,7 +54,7 @@ public function setUp() {
 	public function test_renewal_successful() {
 		// Arrange: Some variables we'll use later.
 		$renewal_order                 = WC_Helper_Order::create_order();
-		$amount                        = 20;
```

Comment:
This change to the test should not be needed. The $20 was there on purpose because WC Subs sends us the amount as well as the order, and I think we should continue to rely on the amount they send for the 4.3 release. That's why I set the amount to be different than the order - to make sure it's tested. I'll add a comment indicating this.

This test wasn't working before in this branch since the amount needs to be used in `generate_create_intent_request `.

This means that the amount would have to be passed in to `create_intent()` and `generate_create_intent_request()`.

Reply:
Ok, having applied this in #955 locally, here's what I think we should do:

`create_intent()` should accept two parameters: the `$renewal_order` and the `$request_array` instead of the `$prepared_source`, like so:

`create_intent($renewal_order,  $request_array)`

This way each function call can pass the `$request_array` it needs without needing to call the child's class `generate_create_intent_request`, and without needing to update the number of arguments we have.